### DISCUSSION TITLE
Wireguard over Speedify?

### DISCUSSION THREAD

**Discussion by netninja71** [SmoothWAN/SmoothWAN#79] [General & Help]
Kind of a weird question, but is it possible to have all LAN traffic routed over Wireguard, with the Wireguard tunnel itself routed over the bonded Speedify connection?

**Comment by TalalMash** [comment]
Hi, sure, use PBR and choose the wireguard port in the entry with Speedify selected.

To add:
The ntp servers were modified from the default openwrt to ip addresses for time synchronization when wireguard is up with no connection, you may have to copy and add them to PBR. Wireguard does not connect without the correct time and may interrupt ntp when up. (default gateway) PBR fixes that. These issues are openwrt related so far, I think few of them were resolved in 22.03

**Reply by netninja71** [reply]
Thank you for this! I’ll do some research on PBR and see if I can figure this out.  I’m glad to know this is even possible, I had been using another router behind my SmoothWAN VM just for the Wireguard tunnel.

**Reply by netninja71** [reply]
> Hi, sure, use PBR and choose the wireguard port in the entry with Speedify selected.
> 
> To add: The ntp servers were modified from the default openwrt to ip addresses for time synchronization when wireguard is up with no connection, you may have to copy and add them to PBR. Wireguard does not connect without the correct time and may interrupt ntp when up. (default gateway) PBR fixes that. These issues are openwrt related so far, I think few of them were resolved in 22.03

Would you be willing to elaborate on this a bit more? I've got the Wireguard tunnel up, and all LAN traffic routed over the Wireguard tunnel, but it doesn't appear that that Wireguard tunnel is traversing the Speedify connection, even thought I've specified it in the PBR policy.  For example, when I run a speedtest across the Wireguard connection, there is no traffic shown on the Speedify traffic graph.  Speedtests also reflect the speed of a single WAN versus the speeds of all 3 WANs bonded.

**Reply by netninja71** [reply]
Just discovered I didn’t have the “Chain” column enabled in the PBR Web GUI, enabled it and changed the chain to “Output” and it’s now working as desired.

**Reply by TalalMash** [reply]
Oh that's interesting, good to hear!
Linked to Wiki.
